presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -645,4 +665,27 @@ private static String getPartitionLocation(Table table, Optional<Partition> part
         }
         return partition.get().getStorage().getLocation();
     }
+
+    private static Optional<ColumnDomain<HiveColumnHandle>> getPathDomain(TupleDomain<HiveColumnHandle> effectivePredicate)
+    {
+        if (!effectivePredicate.getColumnDomains().isPresent()) {
+            return Optional.empty();
+        }
+
+        return effectivePredicate.getColumnDomains().get().stream()
+                .filter(columnDomain -> columnDomain.getColumn().getName().equalsIgnoreCase(PATH_COLUMN_NAME))
+                .findFirst();
+    }
+
+    private static boolean pathMatchesPredicate(TupleDomain<HiveColumnHandle> effectivePredicate,
+            Optional<ColumnDomain<HiveColumnHandle>> pathDomain,
+            String path)
+    {
+        if (!pathDomain.isPresent()) {
+            return true;
+        }
+
+        return effectivePredicate.overlaps(withColumnDomains(
+                ImmutableMap.of(pathDomain.get().getColumn(), singleValue(VARCHAR, utf8Slice(path)))));
```

Comment:
Don't compare a `TupleDomain` and a `path` like this.

Use `Domain.includesNullableValue`.